business/definitionService.js

```diff
@@ -63,7 +64,9 @@ class DefinitionService {
       const curation = this.curationService.get(coordinates, pr)
       return this.compute(coordinates, curation)
     }
-    const existing = force ? null : await this.definitionStore.get(coordinates)
+    const existing = force
+      ? null
+      : (await this.cache.get(this._getCacheKey(coordinates))) || (await this.definitionStore.get(coordinates))
```

Comment:
if it exists and is all ok and is not in the cache, do we want to add it? Otherwise, once something is ejected, it will never get back in and MRU will be broken

Reply:
Ah right. It gets added when it is computed, but this needs to account for it being in the store, but not in cache. Good call